### PR TITLE
[DC-2183] Update string match logic to work with UPPER and lower case.

### PR DESCRIPTION
**Pull request by ksdkalluri**

### CODE REVIEW

File: data_steward/gcloud/gcs/__init__.py [hunk 1]
```diff
@@ -166,14 +170,17 @@ def _get_hpo_bucket_id(self, hpo_id: str) -> str:
...
             dataset_id=LOOKUP_TABLES_DATASET_ID,
             table_id=HPO_ID_BUCKET_NAME_TABLE_ID)
 
+        LOGGER.info(f"Running: {hpo_bucket_query}")
+
         result_df: DataFrame = query(hpo_bucket_query)
-        condition_hpo_id = (result_df['hpo_id'] == hpo_id)
+        condition_hpo_id = ((result_df['hpo_id'].str.lower()) == hpo_id.lower())
         condition_service = (result_df['service'] == service)
         result_filtered = result_df[condition_hpo_id & condition_service]
```
Comment (lrwb-aou) [inline data_steward/gcloud/gcs/__init__.py:185]: @ksdkalluri What is the advantage of selecting everything over what we are specifically looking for?  Wouldn't this work:

```suggestion
        # select query is:
        # select * from `{{project_id}}.{{dataset_id}}.{{table_id}}` where lower(hpo_id) = lower('{{hpo_id}}')
        hpo_bucket_query = JINJA_ENV.from_string(SELECT_QUERY).render(
            project_Id=self.project,
            dataset_id=LOOKUP_TABLES_DATASET_ID,
            table_id=HPO_ID_BUCKET_NAME_TABLE_ID,
            hpo_id=hpo_id)

        result_df: DataFrame = query(hpo_bucket_query)

        if result_df['bucket_name'].count() != 1:
            reutnr 'None'

        return result_df['bucket_name'].iloc[0]
```

To me, the code is shorter and easier to understand.  And I've also been in situations where calling "get all" works fine in test but leads to severe performance issues in production.

Comment (ksdkalluri) [reply data_steward/gcloud/gcs/__init__.py:185]: That is a better approach, I will test it.

File: data_steward/gcloud/gcs/__init__.py [hunk 2]
```diff
@@ -166,18 +170,21 @@ def _get_hpo_bucket_id(self, hpo_id: str) -> str:
...
-            table_id=HPO_ID_BUCKET_NAME_TABLE_ID)
+        LOGGER.info(f"Running: {hpo_bucket_query}")
 
         result_df: DataFrame = query(hpo_bucket_query)
-        condition_hpo_id = (result_df['hpo_id'] == hpo_id)
-        condition_service = (result_df['service'] == service)
-        result_filtered = result_df[condition_hpo_id & condition_service]
 
-        if result_filtered['bucket_name'].count() != 1:
+        LOGGER.info(f"hpo_")
```
Comment (lrwb-aou) [inline data_steward/gcloud/gcs/__init__.py:187]: @ksdkalluri what is this logging?

File: data_steward/constants/utils/bq.py [hunk 3]
```diff
@@ -10,8 +10,14 @@
...
-SELECT_ALL_QUERY = """
-SELECT * FROM `{{project_id}}.{{dataset_id}}.{{table_id}}`
+SELECT_BUCKET_NAME_QUERY = """
+SELECT
+  bucket_name
+FROM
+  `{{project_id}}.{{dataset_id}}.{{table_id}}`
+WHERE
+  LOWER(hpo_id) = LOWER('{{hpo_id}}')
+  AND LOWER(service) = LOWER('{{service}}')
```
Comment (lrwb-aou) [inline data_steward/constants/utils/bq.py:20]: 👍

File: data_steward/gcloud/gcs/__init__.py (hunk 2 above)
Comment (ksdkalluri) [reply data_steward/gcloud/gcs/__init__.py:187]: I am fixing it, didn't realize that was in the code.

**Comment by lrwb-aou** [comment]
Instead of putting `SELECT_ALL_QUERY` back, you could update `https://github.com/all-of-us/curation/blob/develop/data_steward/gcs_utils.py` with the same code.  We know it will be removed when the deprecated module is removed.

**Review by lrwb-aou — APPROVED** [review]
This looks good to me.  All tests have passed.  Thank you for the update and only asking for the data you need.

**Pull request merged**